modules/frontend/frontend.go

```diff
@@ -278,6 +278,33 @@ func newMetricsSummaryHandler(next pipeline.AsyncRoundTripper[combiner.PipelineR
 	})
 }
 
+// cloneChildRequest returns a cloned http.Request from the pipeline.Request.
+func cloneChildRequest(parent pipeline.Request, tenant string, modHTTP func(*http.Request) (*http.Request, error)) (pipeline.Request, error) {
```

Comment:
Since this is a transformation from the original request to the one we will send to the queriers, maybe a more descriptive name like: `cloneRequestForQueriers` could be better

We can also document what modHTTP does